docs/user-guide/node-selection/README.md

```diff
@@ -81,14 +81,96 @@ metadata:
+  nodeSelector:
+	  disktype: ssd
+```
 
 When you then run `kubectl create -f pod.yaml`, the pod will get scheduled on the node that you attached the label to! You can verify that it worked by running `kubectl get pods -o wide` and looking at the "NODE" that the pod was assigned to.
 
+#### Alpha feature in Kubernetes v1.2: Node Affinity
+
+During the first half of 2016 we are rolling out a new mechanism, called *affinity* for controlling which nodes your pods wil be scheduled onto.
+Like `nodeSelector`, affinity is based on labels. But it allows you to write much more expressive rules.
+`nodeSelector` wil continue to work during the transition, but will eventually be deprecated.
+
+Kubernetes v1.2 offers an alpha version of the first piece of the affinity mechanism, called [node affinity](../../design/nodeaffinity.md).
+There are currently two types of node affinity, called `requiredDuringSchedulingIgnoredDuringExecution` and
+`preferresDuringSchedulingIgnoredDuringExecution`. You can think of them as "hard" and "soft" respectively,
+in the sense that the former specifies rules that *must* be met for a pod to schedule onto a node (just like
+`nodeSelector` but using a more expressive syntax), while the latter specifies *preferences* that the scheduler
+will try to enforce but will not guarantee. The "IgnoredDuringExecution" part of the names means that, similar
+to how `nodeSelector` works, if labels on a node change at runtime such that the rules on a pod are no longer
+met, the pod will still continue to run on the node. In the future we plan to offer
+`requiredDuringSchedulingRequiredDuringExecution` which will be just like `requiredDuringSchedulingIgnoredDuringExecution`
+except that it will evict pods from nodes that cease to satisfy the pods' node affinity requirements.
+
+Node affinity is currently expressed using an annotation on Pod. In v1.3 it will use a field, and we will
+also introduce the second piece of the affinity mechanism, called [pod affinity](../../design/podaffinity.md),
+which allows you to control whether a pod schedules onto a particular node based on which other pods are
+running on the node, rather than the labels on the node.
+
+Here's an example of a pod that uses node affinity:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+	name: with-labels
+	annotations:
+		scheduler.alpha.kubernetes.io/affinity: >
+			{
+				"nodeAffinity": {
+					"requiredDuringSchedulingIgnoredDuringExecution": {
+						"nodeSelectorTerms": [
+							{
+								"matchExpressions": [
+									{
+										"key": "kubernetes.io/e2e-az-name",
+										"operator": "In",
+										"values": ["e2e-az1", "e2e-az2"]
+									}
+								]
+							}
+						]
+					},
+					"preferredDuringSchedulingIgnoredDuringExecution": [
+						{
+							"weight": 10,
+							"preference": {"matchExpressions": [
+								{
+									"key": "foo",
+									"operator": "In", "values": ["bar"]
+								}
+							]}
+						 }
+					 ]
+				 }
+			 }
```

Comment:
From line 119 to 149, also need to do
```
s/\t/  /
```
to replace the tab indentation with two spaces